### ISSUE TITLE
Feature: Recipe to select "Best Rate" from multiple exchanges. 

### ISSUE THREAD

**Issue by astupidmoose** [Kukks/btcTransmuter-vnext#5]
Not sure if this is possible, but it might be prudent to allow users to "Shop" the rates from multiple exchanges at the time of sale to ensure they are getting the best rate. 

This would, of course, mean the customer requires BTC holdings on each exchange but might be worth it for high volume merchants where exchanges are vastly different.

**Comment by Kukks** [comment]
Ah, I should have guessed you would be here first with the hard questions
:D let me look into what we can do

**Comment by rockstardev** [comment]
@Kukks if you ask me the bigger priority is #4 - on some exchanges you simply can't execute orders below certain minimum (on Coinbase I think it's $5 or $10)... or you don't want to convert over maximum.

I think for now this feature can remain as brainstorming... for start people will most likely just use whatever exchange is their Rate Source in BtcPayServer

**Comment by Kukks** [comment]
I am working on a new feature to tackle #3 and #4

**Comment by Kukks** [comment]
It's basically a way to create groups of actions that can execute after each other and can pipe data from each action to the next

**Comment by Kukks** [comment]
@astupidmoose  I have now added an action to get the rate of a market symbol on an exchange you integrate.  Still working on the full picture but I can see someone building a recipe action group as follows:

GetExchangeRate: Binance, BTCUSD
GetExchangeRate: Kraken, BTCUSD
GetExchangeRate: TheRockExchange, BTCUSD
ChooseNumericValue: (Minimum, Action1.Bid, Action2.Bid, Action3.Bid)
PlaceOrder( Exchange: Action4, Amount)

**Comment by Kukks** [comment]
at this rate you'll be able to use this as a trading bot lol

**Comment by astupidmoose** [comment]
Yes, that looks awesome! I won't be using as a bot, but I definitely think I would be using it to shop for the best rates.

**Comment by Kukks** [comment]
![tumblr_o16n2kBlpX1ta3qyvo1_1280 (1)](https://user-images.githubusercontent.com/1818366/55616222-9ef15000-5791-11e9-84ed-970bc4d52547.jpg)

This is me for the past few days working on plotting different ways to achieve @astupidmoose 's feature requests
